homeassistant/components/amcrest/camera.py

```diff
@@ -92,12 +106,129 @@ def __init__(self, hass, amcrest):
+        """Flag supported features."""
+        return SUPPORT_ON_OFF
+
+    # Camera property overrides
+
+    @property
+    def is_recording(self):
+        """Return true if the device is recording."""
+        return self._is_recording
+
+    @property
+    def brand(self):
+        """Return the camera brand."""
+        return 'Amcrest'
+
+    @property
+    def model(self):
+        """Return the camera model."""
+        return self._model
+
     @property
     def stream_source(self):
         """Return the source of the stream."""
         return self._camera.rtsp_url(typeno=self._resolution)
+
+    @property
+    def is_on(self):
+        """Return true if on."""
+        return self.video_enabled
+
+    # Additional Amcrest Camera properties
+
+    @property
+    def video_enabled(self):
+        """Return the camera video streaming status."""
+        return self.is_streaming
+
+    @video_enabled.setter
+    def video_enabled(self, enable):
+        """Enable or disable camera video stream."""
+        from amcrest import AmcrestError
+
+        try:
+            self._camera.video_enabled = enable
+        except AmcrestError as error:
+            _LOGGER.error(
+                'Could not %s %s camera video stream due to error: %s',
+                'enable' if enable else 'disable', self.name, error)
+        else:
+            self.is_streaming = enable
+            self.schedule_update_ha_state()
+
+    # Other Entity method overrides
+
+    def update(self):
+        """Update entity status."""
+        from amcrest import AmcrestError
+
+        _LOGGER.debug('Pulling data from %s camera', self.name)
+        try:
+            if self._model is None:
+                self._model = _extract_attr(self._get_cam_attr('device_type'))
+            if not self._static_attrs:
+                self._update_static_attrs()
+            self.is_streaming = self._camera.video_enabled
+            self._is_recording = self._camera.record_mode == 'Manual'
+        except AmcrestError as error:
+            _LOGGER.error(
+                'Could not get %s camera attributes due to error: %s',
+                self.name, error)
+
+    # Other Camera method overrides
+
+    def turn_off(self):
+        """Turn off camera."""
+        self.video_enabled = False
+
+    def turn_on(self):
+        """Turn on camera."""
+        self.video_enabled = True
+
+    # Utility methods
+
+    def _get_cam_attr(self, attr):
+        from amcrest import AmcrestError
+
+        try:
+            return getattr(self._camera, attr)
+        except AmcrestError as error:
+            _LOGGER.error(
+                'Could not get %s camera %s due to error: %s',
+                self.name, attr, error)
+            return None
+
+    def _update_cam_attr(self, attr):
+        value = self._get_cam_attr(attr)
+        if value is not None:
+            self._static_attrs[attr] = _extract_attr(value)
+
+    def _update_static_attrs(self):
```

Comment:
We only want state attributes that users would use in automations and change during the life of the device. None of the elements added in this method belong in `device_state_attributes`.  Some would go into `device_info` if this was loaded via config entry.

Reply:
FWIW, these attributes were already in the sensors, and each of the sensors were retrieving them on every update, resulting in a LOT of unnecessary extra network traffic. This was the first part of removing them from the sensors (which I was planning on doing in my next PR), where they certainly didn't belong. However, if you don't think they should be anywhere, I can remove them here and in sensors.py.

Reply:
They don't belong in state attributes.  If this information needs to be exposed, it can be done through `device_info` after adding support for the device registry.

Reply:
Makes perfect sense. Thanks for the explanation. I'll remove them from both camera.py & sensors.py.